DrDocket/Date.cpp

```diff
@@ -46,13 +46,20 @@ Date
 #include <iostream>
 #include "Date.h"
 
-Date::Date(Month m, unsigned int d, unsigned int y) {
+Month Months[13] = {JAN, JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC};
+
+void Date::initialize(Month m, unsigned int d, unsigned int y) {
     month = m;
     day = (d < 32) ? d : 1;
     year = (y < 2100) ? y : 2019;
     weekNum = findWeekNum(m, day);
     weekday = findWeekDay(m, day);
 }
+
+Date::Date(Month m, unsigned int d, unsigned int y) {
+    this->initialize(m, d, y);
+}
```

Comment:
Why did the constructor need to change? It's not a problem, just curious.

Reply:
I was working on the "parseDatetime" function, which uses date and time objects passed by reference, as a place to store the parsed results.
In that case I needed to call "initialize" on an existing instance, resulting in the same object that the original constructor would have produced.
Factored out the entire body of the constructor so I could use it in two places!